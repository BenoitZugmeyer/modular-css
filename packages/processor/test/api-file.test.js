"use strict";

const namer = require("@modular-css/test-utils/namer.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".file()", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should process a relative file", async () => {
            const result = await processor.file("./packages/processor/test/specimens/simple.css");

            expect(result.exports).toMatchSnapshot();
            expect(result.details.exports).toMatchSnapshot();
            expect(result.details.text).toMatchSnapshot();
            expect(result.details.processed.root.toResult().css).toMatchSnapshot();
        });

        it("should process an absolute file", async () => {
            const result = await processor.file(
                require.resolve("./specimens/simple.css")
            );

            expect(result.exports).toMatchSnapshot();
            expect(result.details.exports).toMatchSnapshot();
            expect(result.details.text).toMatchSnapshot();
            expect(result.details.processed.root.toResult().css).toMatchSnapshot();
        });

        it("should wait for dependencies to be processed before composing", async () => {
            const results = await Promise.all([
                processor.file(require.resolve("./specimens/overlapping/entry1.css")),
                processor.file(require.resolve("./specimens/overlapping/entry2.css")),
            ]);

            expect(results.map((result) => result.exports)).toMatchSnapshot();
        });
    });
});
