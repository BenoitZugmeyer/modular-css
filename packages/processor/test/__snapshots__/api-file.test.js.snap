// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .file() should process a relative file 1`] = `
Object {
  "wooga": Array [
    "wooga",
  ],
}
`;

exports[`/processor.js .file() should process a relative file 2`] = `
Object {
  "wooga": Array [
    "wooga",
  ],
}
`;

exports[`/processor.js .file() should process a relative file 3`] = `
".wooga { color: red; }
"
`;

exports[`/processor.js .file() should process a relative file 4`] = `
".wooga { color: red; }
"
`;

exports[`/processor.js .file() should process an absolute file 1`] = `
Object {
  "wooga": Array [
    "wooga",
  ],
}
`;

exports[`/processor.js .file() should process an absolute file 2`] = `
Object {
  "wooga": Array [
    "wooga",
  ],
}
`;

exports[`/processor.js .file() should process an absolute file 3`] = `
".wooga { color: red; }
"
`;

exports[`/processor.js .file() should process an absolute file 4`] = `
".wooga { color: red; }
"
`;

exports[`/processor.js .file() should wait for dependencies to be processed before composing 1`] = `
Array [
  Object {
    "one": Array [
      "three",
      "two",
      "one",
    ],
  },
  Object {
    "two": Array [
      "three",
      "two",
      "one",
    ],
  },
]
`;
